pandas/io/sas/sas7bdat.py

```diff
@@ -62,12 +63,42 @@ def _convert_datetimes(sas_datetimes: pd.Series, unit: str) -> pd.Series:
             raise ValueError("unit must be 'd' or 's'")
 
 
-class _subheader_pointer:
-    pass
+class _SubheaderPointer:
+    offset: int
+    length: int
+    compression: int
+    ptype: int
```

Comment:
same make this a dataclass

Reply:
these classes are almost entirely unused, in the next pass im going to see if theyre needed at all.  (also want to look at the bytes thing mentioned in a TODO comment).  lets do the dataclass thing then